test/run.d

```diff
@@ -157,51 +215,55 @@ auto predefinedTargets(string[] targets)
                 break;
 
             case "run_runnable_tests", "runnable":
-                newTargets.put(findFiles("runnable"));
+                newTargets.put(findFiles("runnable").map!createDefaultTarget);
                 break;
 
             case "run_fail_compilation_tests", "fail_compilation", "fail":
-                newTargets.put(findFiles("fail_compilation"));
+                newTargets.put(findFiles("fail_compilation").map!createDefaultTarget);
                 break;
 
             case "run_compilable_tests", "compilable", "compile":
-                newTargets.put(findFiles("compilable"));
+                newTargets.put(findFiles("compilable").map!createDefaultTarget);
                 break;
 
             case "all":
+                newTargets ~= createUnitTestTarget();
                 foreach (testDir; testDirs)
-                    newTargets.put(findFiles(testDir));
+                    newTargets.put(findFiles(testDir).map!createDefaultTarget);
+                break;
+            case "unit_tests":
+                newTargets ~= createUnitTestTarget();
                 break;
-
             default:
-                newTargets ~= t;
+                newTargets ~= createDefaultTarget(t);
         }
     }
     return newTargets.data;
 }
 
 // Removes targets that do not need updating (i.e. their .out file exists and is newer than the source file)
-auto filterTargets(string[] targets, string[string] env)
+auto filterTargets(Target[] targets, string[string] env)
 {
     bool error;
     foreach (target; targets)
     {
-        if (!testPath(target).exists)
+        if (!target.exists)
         {
-            writefln("Warning: %s can't be found", target);
+            writefln("Warning: %s can't be found", target.normalizedTestName);
             error = true;
         }
     }
     if (error)
         exit(1);
 
-    string[] targetsThatNeedUpdating;
+    Target[] targetsThatNeedUpdating;
     foreach (t; targets)
     {
-        auto resultRunTime = resultsDir.buildPath(t ~ ".out").timeLastModified.ifThrown(SysTime.init);
-        if (!force && resultRunTime > testPath(t).timeLastModified &&
+        immutable testName = t.normalizedTestName;
+        auto resultRunTime = resultsDir.buildPath(testName ~ ".out").timeLastModified.ifThrown(SysTime.init);
+        if (!force && resultRunTime > testPath(testName).timeLastModified &&
                 resultRunTime > env["DMD"].timeLastModified.ifThrown(SysTime.init))
-            writefln("%s is already up-to-date", t);
+            writefln("%s is already up-to-date", testName);
```

Comment:
Why wouldn't you want the short name here on the CLI?

```
compilable/ddoc9369.d is already up-to-date
compilable/test6056c.d is already up-to-date
compilable/test12807.d is already up-to-date
compilable/test15056.d is already up-to-date
```

Reply:
It gives the same result as on the master branch. I've not changed the behavior, as far as I can see.